src/index.js

```diff
@@ -90,13 +91,36 @@ function callback(err, res) {
       return req._retry();
     }, delay);
   }
-
   const fn = this._callback;
   this.clearTimeout();
 
+  if (!err) {
+    try {
+      if (!this._isResponseOK(res)) {
+        let message = "Unsuccessful HTTP response";
+        if (res) {
+          message = http.STATUS_CODES[res.status] || message;
+        }
+
+        err = new Error(message);
+        err.status = res ? res.status : undefined;
+      }
+    } catch (e) {
+      err = e;
+      err.status = err.status || (res ? res.status : undefined);
```

Comment:
Looks like these two lines are not covered by tests, could you please write a test to check for this outcome? see https://app.codecov.io/gh/luispabon/superagent-retry-delay/pull/14